alkira/resource_alkira_credential_checkpoint_helper.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2022 Alkira Inc. All Rights Reserved.
+package alkira
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+)
+
+func createCheckpointCredential(name, password string, client *alkira.AlkiraClient) (cid string, err error) {
+	c := &alkira.CredentialCheckPointFwService{AdminPassword: password}
+
+	log.Printf("[INFO] Creating Credential (Checkpoint)")
+	return client.CreateCredential(name, alkira.CredentialTypeChkpFw, c)
+}
+
+func createCheckpointCredentialInstances(sicKeys []string, client *alkira.AlkiraClient) error {
+	var err error
+
+	for i, v := range sicKeys {
+		cInstance := &alkira.CredentialCheckPointFwServiceInstance{SicKey: v}
+		log.Printf("[INFO] Creating Credential (CheckpointInstance)")
+
+		name := "checkpoint-instance-credential-" + strconv.Itoa(i)
+		_, credentialErr := client.CreateCredential(name, alkira.CredentialTypeChkpFwInstance, cInstance)
+
+		if credentialErr != nil {
+			err = fmt.Errorf("%w:", err)
+		}
+	}
+
+	return err
+}
+
+func createCheckpointCredentialManagementServer(name, password string, client *alkira.AlkiraClient) error {
+	c := &alkira.CredentialCheckPointFwManagementServer{Password: password}
+
+	log.Printf("[INFO] Creating Credential (Checkpoint Management Server)")
+	_, err := client.CreateCredential(name, alkira.CredentialTypeChkpFwManagement, c)
+
+	return err
+}
+
+func updateCheckpointCredential(cid, name, password string, client *alkira.AlkiraClient) error {
+	c := &alkira.CredentialCheckPointFwService{AdminPassword: password}
+
+	log.Printf("[INFO] Updating Credential (Checkpoint)")
+
+	return client.UpdateCredential(cid, name, alkira.CredentialTypeChkpFw, c)
+}
+
+func updateCheckpointCredentialManagementServerByName(name, password string, client *alkira.AlkiraClient) error {
+	respDetail, err := client.GetCredentialByName(name)
+	if err != nil {
+		return err
+	}
+
+	c := &alkira.CredentialCheckPointFwManagementServer{Password: password}
+	log.Printf("[INFO] Updating Credential (Checkpoint Management Server)")
+
+	return client.UpdateCredential(respDetail.Id, name, alkira.CredentialTypeChkpFwManagement, c)
+}
+
+func deleteCheckpointCredential(cid string, client *alkira.AlkiraClient) error {
+
+	log.Printf("[INFO] Deleting Credential (Checkpoint %s)\n", cid)
+	err := client.DeleteCredential(cid, alkira.CredentialTypeChkpFw)
+
+	if err != nil {
+		log.Printf("[INFO] Credential (Checkpoint %s) was already deleted\n", cid)
+	}
+
+	return nil
+}
+
+func deleteCheckpointCredentialInstances(client *alkira.AlkiraClient) error {
+	var err error
+
+	credentials, err := getAllCheckpointCredentialInstances(client)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] Deleting All Credential (CheckpointInstances)\n")
```

Comment:
This probably won't work.. it may delete other instance credentials as well... I think we should just leave this to backend to handle.

Reply:
No, it won't delete any other credentials. The first thing we do in the function is call and filter credentials for only those that match `chkp-fw-instance` in the credential response. We can't leave this portion up to the backend. Before when we managed instance checkpoint credentials as their own individual resources terraform made the individual calls for us to remove those credential instances. Since we decided to merge all credentials we have to make the calls individually as we clean up.